client/src/components/Applications/AppLayout/AppLayout.js

```diff
@@ -15,64 +15,78 @@ const AppsLayout = appDict => {
-        </h5>
-      </div>
-      {loading && !Object.keys(categoryDict).length ? (
+      {loading && !hasCategoryDict ? (
         <LoadingSpinner />
       ) : (
         <>
-          {Boolean(Object.keys(categoryDict).length) && <AppBrowser />}
-          {!params.appId && (
-            <AppPlaceholder apps={Boolean(Object.keys(categoryDict).length)} />
-          )}
+          {hasCategoryDict && <AppBrowser />}
+          {!params.appId && <AppPlaceholder apps={hasCategoryDict} />}
         </>
       )}
     </>
   );
 };
 
+const AppsHeader = appDict => {
+  const { params } = useRouteMatch();
+  const appMeta = appDict[params.appId];
+  const path = appMeta ? ` / ${appMeta.value.definition.label}` : '';
+
+  return `Applications ${path}`;
+};
+
 const AppsRoutes = () => {
   const { path } = useRouteMatch();
   const dispatch = useDispatch();
   const appDict = useSelector(state => state.apps.appDict, shallowEqual);
+  const hasAppDict = Boolean(Object.keys(appDict).length);
 
   return (
-    <div id="apps-wrapper">
-      <Route path={`${path}/:appId?`}>
-        <AppsLayout appDict={appDict} />
-      </Route>
-      {Object.keys(appDict).length ? (
-        <Route
-          exact
-          path={`${path}/:appId`}
-          render={({ match: { params } }) => {
-            const appDef = appDict[params.appId];
-            if (appDef && 'html' in appDef) {
-              dispatch({
-                type: 'LOAD_APP',
-                payload: appDict[params.appId]
-              });
-            } else {
-              dispatch({
-                type: 'GET_APP',
-                payload: {
-                  appId: params.appId
+    <Section
+      bodyClassName="has-loaded-applications"
+      routeName="APPLICATIONS"
+      header={
+        <Route path={`${path}/:appId?`}>
+          <AppsHeader appDict={appDict} />
+        </Route>
+      }
+      content={
+        <>
+          <Route path={`${path}/:appId?`}>
+            <AppsLayout />
+          </Route>
+          {hasAppDict ? (
+            <Route
+              exact
+              path={`${path}/:appId`}
+              render={({ match: { params } }) => {
+                const appDef = appDict[params.appId];
+                if (appDef && 'html' in appDef) {
+                  dispatch({
+                    type: 'LOAD_APP',
+                    payload: appDict[params.appId]
+                  });
+                } else {
+                  dispatch({
+                    type: 'GET_APP',
+                    payload: {
+                      appId: params.appId
+                    }
+                  });
                 }
-              });
-            }
-            return <AppDetail />;
-          }}
-        />
-      ) : null}
-    </div>
+                return <AppDetail />;
+              }}
+            />
+          ) : null}
+        </>
+      }
+      contentClassName="apps-content"
+      contentLayoutName="oneRow"
+    />
```

Comment:
- Pass `appDict` to new `AppsHeader` instead of `AppsLayout`, because:
    - The `appDict` is only used in the header.
    - The header is isolated in `AppsHeader`.
- Replace `Object.keys(appDict).length` with `hasAppDict` to:
    - Be consistent with `hasCategoryDict`.
    - Abstract the logic that sets the value from the use of the value.
- Apply `Section` component.